dbms/src/Functions/geohashCoverBox.cpp

```diff
@@ -0,0 +1,166 @@
+        validateArgumentType(*this, arguments, 4, isUInt8, "integer");
+
+        if (!(isSame(arguments[0], arguments[1]) &&
+              isSame(arguments[0], arguments[2]) &&
+              isSame(arguments[0], arguments[3])))
+        {
+            throw Exception("Illegal type of argument of " + getName() +
+                            " all coordinate arguments must have the same type, instead they are:" +
+                            arguments[0]->getName() + ", " +
+                            arguments[1]->getName() + ", " +
+                            arguments[2]->getName() + ", " +
+                            arguments[3]->getName() + ".",
+                            ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+        }
+
+        return std::make_shared<DataTypeArray>(std::make_shared<DataTypeString>());
+    }
+
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    template <typename LonMinType, typename LatMinType, typename LonMaxType, typename LatMaxType, typename PrecisionType>
+    bool tryExecute(const IColumn * lon_min_column,
+                    const IColumn * lat_min_column,
+                    const IColumn * lon_max_column,
+                    const IColumn * lat_max_column,
+                    const IColumn * precision_column,
+                    ColumnPtr & result)
+    {
+        static constexpr size_t max_elements = 10'000'000;
+
+        const auto * lon_min = checkAndGetColumn<ColumnVector<LonMinType>>(lon_min_column);
+        const auto * lat_min = checkAndGetColumn<ColumnVector<LatMinType>>(lat_min_column);
+        const auto * lon_max = checkAndGetColumn<ColumnVector<LonMaxType>>(lon_max_column);
+        const auto * lat_max = checkAndGetColumn<ColumnVector<LatMaxType>>(lat_max_column);
+        const auto * precision = checkAndGetColumn<ColumnVector<PrecisionType>>(precision_column);
+
+        if (!lon_min || !lat_min || !lon_max || !lat_max)
+            return false;
+
+        const size_t total_rows = lat_min->size();
+
+        auto col_res = ColumnArray::create(ColumnString::create());
+        ColumnString & res_strings = typeid_cast<ColumnString &>(col_res->getData());
+        ColumnArray::Offsets & res_offsets = col_res->getOffsets();
+        ColumnString::Chars & res_strings_chars = res_strings.getChars();
+        ColumnString::Offsets & res_strings_offsets = res_strings.getOffsets();
+
+        UInt32 pos = 0;
+        for (size_t row = 0; row < total_rows; ++row)
+        {
+            const Float64 lon_min_value = lon_min->getElement(row);
+            const Float64 lat_min_value = lat_min->getElement(row);
+            const Float64 lon_max_value = lon_max->getElement(row);
+            const Float64 lat_max_value = lat_max->getElement(row);
+            const PrecisionType precision_value = GeoUtils::geohashPrecision(precision->getElement(row));
+
+            const auto positions = GeoUtils::geohashCoverBox(lon_min_value, lat_min_value,
```

Comment:
Can we skip the intermediate vector here, and construct the resulting array right away?

Reply:
Well, I think it is possible. Let me try

Reply:
Done